app/models/cell_metadatum.rb

```diff
@@ -43,23 +43,26 @@ class CellMetadatum
 
   # concatenate all the necessary data_array objects and construct a hash of cell names => expression values
   def cell_annotations
-    cells = self.study.all_cells_array
+    cells = self.all_study_cells
     # replace blank/nil values with default missing label
     annot_values = AnnotationVizService.sanitize_values_array(
       self.concatenate_data_arrays(self.name, 'annotations'), self.annotation_type
     )
     Hash[cells.zip(annot_values)]
   end
 
-  # concatenate data arrays of a given name/type in order
+  def all_study_cells
+    query = {name: 'All Cells', array_type: 'cells', linear_data_type: 'Study', linear_data_id: self.study_id,
+             cluster_name: self.study_file.upload_file_name, study_id: self.study_id, study_file_id: self.study_file_id,
+             cluster_group_id: nil, subsample_annotation: nil, subsample_threshold: nil}
+    DataArray.where(query).order(:array_index => 'asc').pluck(:values).reduce([], :+)
+  end
```

Comment:
Let's talk through this Monday -- I like that we're able to utilize more of the index fields.  I wonder how much of this we can port over to the method in `study.rb`. Theoretically, this is doing exactly the same thing, and study annotations are always defined in the same study_file (the metadata file), so we should be able to get the performance gains you've gained here, but keep the all_cells_array method in study.rb where it makes more sense

Reply:
Happy to talk about it - but what I had noticed was that part of the slowdown was using the association (i.e. `self.study.all_cells_array`) to call the method.